utils/snapshot_extractor.mjs

```diff
@@ -30,7 +30,8 @@ export default async function snapshotExtractor(extractor, { inputs }) {
 
   let writeResult = "";
   const write = (string) => {
-    writeResult += string + "\n";
+    if (extractor.write) writeResult = extractor.write(writeResult, string);
```

Comment:
If we really want to make this kind of change, then we also have to add this condition in lifecycle.mjs. But I'm not sure if it is necessary. Generally we want to do as little "transformative" tasks in an extractor strategy as possible. Consider this: Any data source we're extracting data from could potentially always change their business logic and so for extraction strategies what we want to primarily focus on is extracting data and writing it as neutrally as possible to the disk without the risk of e.g. parsing the data and then having to error.

And then once we've successfully run an extractor strategy, we want to bundle all of this risk about transformation in a transformer strategy that is fast to run (because it works purely on disk) and because we can then make the edits there.

So I highly recommend you separate your code into:

- retrieving and directly writing raw data through networking => extractor strategy
- parsing, mapping and editing data => transformation

Reply:
Okay, then I will implement this part in transformation